File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.js
function solution(numbers) {
    var answer = '';
    var nums = numbers.join(' ').split(" ")
    
    nums.sort((a,b)=>(b+a)-(a+b))
    
    answer = nums.join("")
    
    //return answer;
    
    return answer[0]==='0'? '0' : answer;
}

File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.py
# def solution(numbers):
#     answer = []
#     num_list = []
#     numbers.sort(reverse=True)
#     numbers=str(numbers)
    
#     print(numbers)
#     num_list = sorted(numbers, key=lambda x:x)
#     # print(num_list)
    
#     print(num_list)
#     return answer

from itertools import permutations
def solution(numbers):
    num_length = len(numbers)
    numbers = list(map(str, numbers))
    
    #result = max(set(("".join(x)) for x in list(permutations(numbers, num_length))))
    result = max(("".join(x)) for x in list(permutations(numbers, num_length)))

    return result 